src/pururu/domain/services/pururu_service.py

```diff
@@ -58,6 +58,44 @@ def retrieve_player_stats(self, player: str) -> MemberStats:
                         member_stats.justifications += 1
                         member_stats.points += 1
         return member_stats
+    
+    def retrieve_montly_report(self, month_offset: int) -> MonthlyReport:
+        """
+        Retrieves the montly report
+        :param month_offset: month position relative to current month
+        :return: MonthlyReport
+        """
+        self.logger.debug(f"Retrieving stats for month offset {month_offset}")
+        attendances = self.database_service.get_all_attendances()
+        current_month = datetime.now().month
+        correct_month = current_month+month_offset
+        total_events = 0
+        membersDict = dict()
+        gamesDict = dict()
+        for attendance in attendances:
+            month = 0
+            if ":" not in attendance.date:
+                date = datetime.strptime(attendance.date, "%Y-%m-%d")
+                month = date.month
+            else:
+                date = datetime.strptime(attendance.date, "%Y-%m-%d %H:%M:%S")
+                month = date.month
+            if month != correct_month:
+                continue
+            for member in attendance.members:
+               if member.member not in membersDict:
+                   membersDict[member.member] = dict(attendance = 0, justified = 0)
+               if member.attendance:
+                   membersDict[member.member]["attendance"] += 1
+               if member.justified:
+                   membersDict[member.member]["justified"] += 1
+            if attendance.game_id not in gamesDict:
+                gamesDict[attendance.game_id] = 0
```

Comment:
This part is weird, game_ids are unique so you will end up having:
```python
gamesDict[4] = 1
gamesDict[5] = 1
...
gamesDict[X] = 1
```
I think is best to just have a list of the game_ids